gslb/ingestion/gslb_host_rule.go

```diff
@@ -0,0 +1,272 @@
+	gslbhrName := gslbhr.ObjectMeta.Name
+	gslbhrSpec := gslbhr.Spec
+	if gslbhrSpec.Fqdn == "" {
+		return errors.New("GSFqdn missing for " + gslbhrName + " GSLBHostRule")
+	}
+	//TTL needs to be an integer
+	if reflect.TypeOf(gslbhrSpec.TTL).Kind() != reflect.Int {
+		return errors.New("value of TTL should an integer for " + gslbhrName + " GSLBHostRule")
+	}
+
+	sitePersistences := gslbhrSpec.SitePersistence
+	for _, site := range sitePersistences {
+		if reflect.TypeOf(site.Enabled).Kind() != reflect.Bool {
+			return errors.New("Site Persistence enable value should be a bool for " + gslbhrName + " GSLBHostRule")
+		}
+		sitePersistenceProfileName := site.ProfileRef
+		if site.Enabled == true && isSitePersistenceProfilePresent(sitePersistenceProfileName) != true {
+			return errors.New("SitePersistence Profile " + sitePersistenceProfileName + " error for " + gslbhrName + " GSLBHostRule")
+		}
+	}
+
+	thirdPartyMembers := gslbhrSpec.ThirdPartyMembers
+	gslbutils.Logf("Verifying thirdPartyMembers!!!")
+	for _, tpmember := range thirdPartyMembers {
+		if vip := net.ParseIP(tpmember.VIP); vip == nil {
+			return errors.New("Invalid VIP for thirdPartyMember site " + tpmember.Site + ", " + gslbhrName + " GSLBHostRule (expecting IP address")
+		}
+		if isThirdPartyMemberSitePresent(tpmember.Site) != true {
+			return errors.New("ThirdPartyMember site " + tpmember.Site + " doesnot exist for " + gslbhrName + " GSLBHostRule")
+		}
+		gslbutils.Logf("Verified thirdPartyMembers!!!")
+	}
+	gslbutils.Logf("Verified thirdPartyMembers!!!")
+
+	healthMonitorRefs := gslbhrSpec.HealthMonitorRefs
+	for _, ref := range healthMonitorRefs {
+		if isHealthMonitorRefPresent(ref) != true {
+			return errors.New("Health Monitor Ref " + ref + " error for " + gslbhrName + " GSLBHostRule")
+		}
+	}
+
+	for _, tp := range gslbhrSpec.TrafficSplit {
+		if tp.Weight < 1 || tp.Weight > 20 {
+			return errors.New("traffic weight " + strconv.Itoa(int(tp.Weight)) + " must be between 1 and 20")
+		}
+	}
+	return nil
+}
+
+func AddGSLBHostRuleObj(obj interface{}) {
+	gslbhr, ok := obj.(*gslbhralphav1.GSLBHostRule)
+	if !ok {
+		gslbutils.Errf("object added is not of type GSLB Host Rule")
+		return
+	}
+
+	// GSLBHostRule for all other namespaces are rejected
+	if gslbhr.ObjectMeta.Namespace != gslbutils.AVISystem {
+		return
+	}
+
+	//Validate GSLBHostRule
+	err := ValidateGSLBHostRule(gslbhr)
+	if err != nil {
+		gslbutils.Errf("Error in accepting GSLB Host Rule %s : %s", gslbhr.ObjectMeta.Name, err.Error())
+		return
+	}
+
+	gslbutils.Logf("ns: %s, gslbhostrule: %s, msg: %s", gslbhr.ObjectMeta.Namespace, gslbhr.ObjectMeta.Name,
+		"GSLBHostRule object added")
+}
+
+func UpdateGSLBHostRuleObj(old, new interface{}) {
+	oldGslbhr := old.(*gslbhralphav1.GSLBHostRule)
+	newGslbhr := new.(*gslbhralphav1.GSLBHostRule)
+	if oldGslbhr.ObjectMeta.ResourceVersion == newGslbhr.ObjectMeta.ResourceVersion {
+		return
+	}
+	if oldGslbhr.ObjectMeta.Namespace != newGslbhr.ObjectMeta.Namespace {
```

Comment:
I don't think the `Namespace` is editable for an object once it is created. If there's no specific reason for this check, we can remove this.